configure.ac

```diff
@@ -80,26 +80,46 @@ AC_ARG_WITH(gtk,
     [ac_with_gtk="${withval}"],
     [ac_with_gtk=2])
 
-# Checks for libraries.
-
-PKG_CHECK_MODULES(XLIB, "x11")
-AC_SUBST(XLIB_CFLAGS)
-AC_SUBST(XLIB_LIBS)
+AC_ARG_ENABLE(wayland,
+    [AC_HELP_STRING([--enable-wayland],
+        [Enable Wayland support])],
+    [enable_wayland="${enableval}"],
+    [enable_wayland=no])
+
+AC_ARG_ENABLE(x11,
+    [AC_HELP_STRING([--enable-x11],
+        [Enable X11 support])],
+    [enable_x11="${enableval}"],
+    [enable_x11=yes])
+
+if test "x$enable_wayland" = "xno" && test "x&$enable_x11" = "xno"; then
+    AC_MSG_ERROR([At least one of X11 and Wayland needs to be enabled])
+fi
 
-gio_modules="gthread-2.0 gio-unix-2.0 >= 2.18.0"
-PKG_CHECK_MODULES(GIO, [$gio_modules])
-AC_SUBST(GIO_CFLAGS)
```

Comment:
Why do you think it's better to not separate `GIO_FLAGS` from the rest?

Reply:
Currently we were calling e.g. PKG_CHECK_MODULES twice for those Gio packages and we aren't using just GIO_CFLAGS in a Makefile.am, but always as part of FM_CFLAGS, so I thought it might as well just be part of the rest.